### ISSUE TITLE
Travis is broken

### ISSUE THREAD

**Issue by traversaro** [robotology/human-dynamics-estimation#124]
On the master branch, the Travis compilation tests are broken. It has been like that for a long time, but I am opening an issue to track this.

**Comment by traversaro** [comment]
If there is no plan to fix this in the short or medium term, I suggest to just go ahead and deactivate Travis. Wasting 3.5 minutes for 8 builds for each commit, on builds that are known to be broken, is removing useful resources to the rest of the robotology projects that use Travis (Travis resources are shared inside an organization).

**Comment by lrapetti** [comment]
Addressed in https://github.com/robotology/human-dynamics-estimation/pull/125.

**Comment by traversaro** [comment]
Note that the issue is still present for the master branch, cc @DanielePucci .

**Comment by lrapetti** [comment]
I can cherry pick the same commit and apply also in `master`

**Comment by lrapetti** [comment]
> I can cherry pick the same commit and apply also in `master`

https://github.com/robotology/human-dynamics-estimation/pull/127, I am waiting for the checks before merging.

**Comment by lrapetti** [comment]
Travis is now fixed for Linux by removing the support for `Ubuntu 16.04`.

For MacOS, the failure described in https://github.com/robotology/human-dynamics-estimation/pull/128#issuecomment-500768108 has to be addressed.

**Comment by lrapetti** [comment]
> Travis is now fixed for Linux by removing the support for `Ubuntu 16.04`.
> 
> For MacOS, the failure described in [#128 (comment)](https://github.com/robotology/human-dynamics-estimation/pull/128#issuecomment-500768108) has to be addressed.

I think the reason of this failure is the fact we are not installing `icub-main` in the CI environment, but it is used in the [`RobotPositionController.cpp`](https://github.com/robotology/human-dynamics-estimation/blob/master/devices/RobotPositionController/RobotPositionController.cpp#L22). I will try to add it and check if it fixies Travis.

**Comment by traversaro** [comment]
Mhh, how does that work without `find_package(ICUB REQUIRED)` and properly linking the iCub  libraries?

**Comment by lrapetti** [comment]
> Mhh, how does that work without `find_package(ICUB REQUIRED)` and properly linking the iCub libraries?

I was wondering the same, it is not present. But the [`ctrl`](https://github.com/robotology/human-dynamics-estimation/blob/devel/devices/RobotPositionController/CMakeLists.txt#L28) library should be in `icub-main`, so it should be somehow linked in our machine since everyone is able to compile it.

**Comment by traversaro** [comment]
I am not really sure what you mean by "linked in our machine" (see https://github.com/green7ea/cpp-compilation#a-linkers-job and section 2.2 of https://www.dropbox.com/s/nsimhycn8bgherf/meson_sample_chapter.pdf for more info on "linking").

**Comment by lrapetti** [comment]
> I am not really sure what you mean by "linked in our machine"

I meant that in our machine (HDE users), we are don't get any compilation problem so the ctrl library is linked.

**Comment by traversaro** [comment]
> I meant that in our machine (HDE users), we are don't get any compilation problem so the ctrl library is linked.

Ack, I was confused because linking is something done at the executable/library level and not at the machine level, now I see what you mean.

**Comment by lrapetti** [comment]
I have open a PR https://github.com/robotology/human-dynamics-estimation/pull/129 to check if adding `icub-main` solves the problem.

**Comment by lrapetti** [comment]
> I have open a PR #129 to check if adding `icub-main` solves the problem.

Apparently, it solved the problem since at least one of the two MacOs environment finished correctly. However, I just add the build and install step for `icub-main`, @traversaro  do you think I should also add `find_package(ICUB REQUIRED)` in order to have a clear error when `icub-main` is not installed?

**Comment by traversaro** [comment]
> However, I just add the build and install step for icub-main, @traversaro do you think I should also add find_package(ICUB REQUIRED) in order to have a clear error when icub-main is not installed?

Yes. You just need to add `find_package(ICUB REQUIRED)` as actually the linking is already provided by this line: https://github.com/robotology/human-dynamics-estimation/blob/8cfc88c8fd9fad41ebe837e1c91c59f91df376b9/devices/RobotPositionController/CMakeLists.txt#L28 that I did not noticed before. Note that we did not noticed this problem before because in icub-main we are not exporting the CMake targets with a namespace, that would permit to catch this problems (use of a namespaced target in `target_link_libraries` more easily).
